spec/features/providers/details_spec.rb

```diff
@@ -44,19 +56,37 @@
     end
   end
 
-  def test_details_page(expected_status)
-    stub_api_v2_request(
-      "/recruitment_cycles/#{provider.recruitment_cycle.year}",
-      provider.recruitment_cycle.to_jsonapi
-    )
-    stub_api_v2_request(
-      "/recruitment_cycles/#{provider.recruitment_cycle.year}" \
-        "/providers/#{provider.provider_code}",
-      provider.to_jsonapi
-    )
+  context "Current recruitment cycle" do
+    let(:provider) do
+      build :provider,
+            provider_code: 'AO',
+            content_status: 'draft',
+            recruitment_cycle: build(:recruitment_cycle, year: Settings.current_cycle)
+    end
 
-    visit details_provider_recruitment_cycle_path(provider.provider_code, provider.recruitment_cycle.year)
+    it 'Displays the publish button' do
+      expect(org_detail_page.publish_button).to be_present
+    end
+  end
 
+  context "Next recruitment cycle" do
+    let(:provider) do
+      build :provider,
+            provider_code: 'AO',
+            content_status: 'draft',
+            recruitment_cycle: build(:recruitment_cycle, year: Settings.current_cycle + 1)
```

Comment:
Same comment as above to just use the default factory values instead of micro-managing them. :)